### ISSUE TITLE
R can't find packages installed by travis

### ISSUE THREAD

**Issue by zachmayer** [travis-ci/travis-ci#4641]
@craigcitro and @hadley 
We're trying to add some unit tests to the [caret package][1] that get run by travis, but not on CRAN.  This saves build time on CRAN and reduces the number of dependencies they have to install to check our package, while letting us run a more complete test suite on travis.

I thought I could simply install the requirements for the test using the `r_packages:` line in my [travis.yml file][2]:

```{yaml}
    r_packages:
     - ROSE
     - DMwR
```

[However, my `NOT_CRAN=TRUE` builds are still failing][3].  (NOT_CRAN=`FALSE` runs fine as the problematic tests are skipped)

This is really strange, as [when I look at the build logs][4], I see travis successfully installing all the packages I need:

```{bash}
    * installing *source* package â€˜ROSEâ€™ ...
    ** package â€˜ROSEâ€™ successfully unpacked and MD5 sums checked
    ** R
    ** data
    ** inst
    ** preparing package for lazy loading
    ** help
    *** installing help indices
    ** building package indices
    ** testing if installed package can be loaded
    * DONE (ROSE)
```

But when my tests run that depend on those packages, R can't find them:

```{R}
    > library(testthat)
    > library(caret)
    Loading required package: lattice
    Loading required package: ggplot2
    > 
    > test_check("caret")
    [1] "Reduced dimension to 3 by default. "
    1 package is needed for this model and is not installed. (ROSE). Would you like to try to install it now?1. Error: check appropriate sampling calls by name -----------------------------
    
    1: withCallingHandlers(eval(code, new_test_environment), error = capture_calls, message = function(c) invokeRestart("muffleMessage"), 
           warning = function(c) invokeRestart("muffleWarning"))
    2: eval(code, new_test_environment)
    3: eval(expr, envir, enclos)
    4: caret:::parse_sampling(i) at test_sampling_options.R:14
    5: checkInstall(pkgs)
    6: stop()
    
    testthat results ================================================================
    OK: 62 SKIPPED: 0 FAILED: 1
    1. Error: check appropriate sampling calls by name 
    
    Error: testthat unit tests failed
    Execution halted
```

(I think) the [relevant line of code is here in caret's source code][5]:

```{R}
    good <- rep(TRUE, length(pkg))
      for(i in seq(along = pkg)){
        tested <- try(find.package(pkg[i]), silent = TRUE)
        if(class(tested)[1] == "try-error") good[i] <- FALSE
      }
```

Why can't the `find.package` function find packages installed by travis?  Do they go in a special, separate library somewhere?

Also, as an aside, how do I make my travis builds for r less verbose?  By default they seem to print way too much information (e.g. it echoes all code run by the tests and manual, even code that doesn't error).

  [1]: https://github.com/topepo/caret
  [2]: https://github.com/topepo/caret/blob/master/.travis.yml
  [3]: https://travis-ci.org/topepo/caret/builds/74496373
  [4]: https://s3.amazonaws.com/archive.travis-ci.org/jobs/74496374/log.txt
  [5]: https://github.com/topepo/caret/blob/593fff8f27034eec290b311c63c01cb6cd763d22/pkg/caret/R/modelLookup.R#L4

**Comment by zachmayer** [comment]
(Copied from stack overflow](http://stackoverflow.com/q/31879011/345660))

**Comment by zachmayer** [comment]
P.S.: I tried to add the `community:r` label, but couldn't.

**Comment by craigcitro** [comment]
the install scripts for travis aren't doing anything special -- they use devtools, but they ultimately call into `install.packages` to do the actual package installation.

does `caret` play any games with library paths or installing its own packages?

**Comment by zachmayer** [comment]
Nope.  But it seems "R CMD CHECK --as-cran" does play games with the library paths and sets library to `/usr/lib/R/library`.  ([See here on stackoverflow](http://stackoverflow.com/a/31932468/345660))

**Comment by hadley** [comment]
During `R CMD check`, only packages explicitly listed in the DESCRIPTION are made available, so unfortunately your basic approach won't work.
